src/Storages/Distributed/DistributedSink.cpp

```diff
@@ -378,9 +378,7 @@ DistributedSink::runWritingJob(JobReplica & job, const Block & current_block, si
                     /// (anyway fallback_to_stale_replicas_for_distributed_queries=true by default)
```

Comment:
Can you make a deep copy of settings here, by the way? At least that can help guarantee that we have the same settings during the initialisation and the final check